components/suggest/src/db.rs

```diff
@@ -905,7 +905,7 @@ impl<'a> SuggestDao<'a> {
                   s.provider = ?
                   AND k.keyword = ?
                   AND d.suggestion_type IN ({})
-                  AND NOT EXISTS (SELECT 1 FROM dismissed_suggestions WHERE url = s.url)
+                  AND NOT EXISTS (SELECT 1 FROM dismissed_dynamic_suggestions WHERE dismissal_key = s.url AND suggestion_type = d.suggestion_type)
```

Comment:
Nit: This is a little long, so could you wrap it somehow please? We don't have a totally consistent style though. I would do something like:

```
                  AND NOT EXISTS (
                      SELECT 1 FROM dismissed_dynamic_suggestions
                      WHERE dismissal_key = s.url AND suggestion_type = d.suggestion_type
                  )
```